src/edu/washington/escience/myria/operator/agg/AggUtils.java

```diff
@@ -84,14 +88,19 @@ public static boolean needsStats(final Set<AggregationOp> aggOps) {
    *
    * @param factories The factories that will produce the aggregators.
    * @param inputSchema The schema of the input tuples.
+   * @param pyFuncRegistrar - python function registrar to get functions from workers.
    * @return the aggregators for this operator.
    * @throws DbException if there is an error.
    */
   public static Aggregator[] allocateAggs(
-      final AggregatorFactory[] factories, final Schema inputSchema) throws DbException {
+      final AggregatorFactory[] factories,
+      final Schema inputSchema,
+      final PythonFunctionRegistrar pyFuncRegistrar)
```

Comment:
It's only required by `PythonUDFEvaluator`. What you only need to do is to get it from `Worker` during initialization of `PythonUDFEvaluator`, then you can save many of such additional arguments.